lib/Conversion/TritonGPUToLLVM/ReduceOpToLLVM.cpp

```diff
@@ -367,8 +367,10 @@ struct ReduceOpConversion
     Value warpId = udiv(threadId, warpSize);
     Value laneId = urem(threadId, warpSize);
 
-    auto threadsPerWarp = triton::gpu::getThreadsPerWarp(srcLayout);
-    auto warpsPerCTA = triton::gpu::getWarpsPerCTA(srcLayout);
+    auto threadsPerWarp =
```

Comment:
cc @zahimoud, what I wanted is still the original shape->`[2, 2]` but not `[2, 1]` because I wanted to know the exact warp distribution that matches the parent layout. But I found it's tricky to modify handle the parent `shape` using the child `order`, so I'll just reply on `getThreadsPerWarpWithUniqueData` and insert an assert for `getWarpsPerCTA` to prevent future misuse.

Reply:
Why can't we use the getWarpsPerCTA for the parent layout instead ? Yes the assert is needed there, thanks.

Reply:
Because the column dimension indeed only has two "logical" warps, the data are not distributed over four warps.

Reply:
Oh I see, so basically you want the getWarpsPerCTAWithUniqueData with 1 inserted on the sliced dim ?

Reply:
Nope, I think what I want is `getWarpsPerCTA` but returns [2, 2], currently it returns [4, 1]. However, after thinking about it a bit, I think returning [2, 2] is also confusing...So I'd better just use `getThreadsPerWarpWithUniqueData` and get around it by modifying the `delinearize` function.